src/components/PageSizeSelector/usePageSizeSelector.ts

```diff
@@ -0,0 +1,14 @@
+import { RECENT_PAGE_SIZE_KEY } from "constants/index";
+import { useUpdateURLQueryParams } from "hooks";
+
+/** usePageSizeSelector updates the page size query param and saves the page size to local storage */
+const usePageSizeSelector = () => {
+  const updateURLQueryParams = useUpdateURLQueryParams();
+  const setPageSize = (pageSize: number) => {
+    localStorage.setItem(RECENT_PAGE_SIZE_KEY, `${pageSize}`);
+    updateURLQueryParams({ limit: pageSize.toString(), page: "0" });
```

Comment:
[nit] Since `${pageSize}` and `pageSize.toString()` do the same thing, you could stringify pageSize once and use the variable for both